app/controllers/events_controller.rb

```diff
@@ -29,11 +29,13 @@ def new
 
   def create
     @event = current_user.events.build(event_params)
-    if @event.save
+    if @event.only_woman && current_user.gender != 'woman'
```

Comment:
```suggestion
    if @event.only_woman && !current_user.woman?
```
enumで追加されるメソッドでシンプルに書けそうです。